tools/test.py

```diff
@@ -740,6 +743,12 @@ def Execute(args, context, timeout=None, env={}, faketty=False):
   for key, value in env.iteritems():
     env_copy[key] = value
 
+  preexec_fn = None
+  if disable_core_files:
+    def disableCoreFiles():
+      resource.setrlimit(resource.RLIMIT_CORE, (0,0))
+    preexec_fn = disableCoreFiles
```

Comment:
you could simply do:
```python
    preexec_fn = lambda: resource.setrlimit(resource.RLIMIT_CORE, (0,0))
```
or even:
```python
  preexec_fn = lambda: resource.setrlimit(resource.RLIMIT_CORE, (0,0)) if disable_core_files else None
```
instead of L746-L750

Reply:
The second form might be misleading. If we are doing it, the function has to be covered in parens.

Reply:
Nice, thanks. Done.